plugins/woocommerce/tests/e2e-pw/tests/shopper/checkout.spec.js

```diff
@@ -105,9 +98,23 @@ test.describe( 'Checkout page', () => {
 		}
 	} );
 
-	test.beforeEach( async ( { context } ) => {
+	test.beforeEach( async ( { context, baseURL } ) => {
 		// Shopping cart is very sensitive to cookies, so be explicit
 		await context.clearCookies();
+		const api = new wcApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
+		// enable bank transfers and COD for payment
+		await api.put( 'payment_gateways/bacs', {
+			enabled: true,
+		} );
+		await api.put( 'payment_gateways/cod', {
+			enabled: true,
+		} );
+
```

Comment:
I moved these from `beforeAll()` to `beforeEach()` because we were seeing some failures where no payment option is available (I think other tests were disabling them).  Enabling them before each test minimizes the chance of it being disabled.

Reply:
This is one is interesting! I agree that handling cookies individually per test sounds safer than doing it per `describe` block. By any chance, did you compare the test suite duration of the previous implementation with `beforeAll` with the new one? It would be nice to know if the performance has been penalised.

Reply:
I didn't benchmark it, but that call is so fast (it's just toggling a single field in the settings table) that I can't imagine that it has a big performance impact.  That is a great point to consider in the future though -- making sure that the right things are going into `beforeAll()` and `beforeEach()`!